src/code/reducers/index.js

```diff
@@ -73,6 +75,33 @@ export default function reducer(state, action) {
         challengeProgress: progress
       });
     }
+    case actionTypes.EGG_SELECTED: {
+      return state.merge({
+        selectedEggIndex: action.eggIndex
+      });
+    }
+    case actionTypes.EGG_SUBMITTED: {
+      let progress = updateProgress(state, action.correct);
+
+      return state.merge({
+        submittedEggIndex: action.eggIndex,
+        submittedBasketIndex: action.basketIndex,
+        success: action.correct,
+        challengeProgress: progress
+      });
+    }
+    case actionTypes.EGG_REJECTED: {
+      return state.merge({
+        submittedEggIndex: null,
+        submittedBasketIndex: null
+      });
+    }
+    case actionTypes.EGG_ACCEPTED: {
+      return state.merge({
+        submittedEggIndex: null,
+        submittedBasketIndex: null
+      });
+    }
```

Comment:
Done. I like the way the passing arguments to actions through the modal dialog turned out. The other use of the `submittedEggIndex` and `submittedBasketIndex` properties was to trigger post-dialog animations, which is now handled through more generally useful `correct` and `errors` counters.